specification/sbi_atee.adoc

```diff
@@ -13,6 +44,953 @@ running a VMM).
+    TSM_NOT_LOADED = 0,
+    /* TSM has been loaded, but has not yet been initialized. */
+    TSM_LOADED = 1,
+    /* TSM has been loaded & initialized, and is ready to accept ECALLs.*/
+    TSM_READY = 2
+};
+
+struct tsm_info {
+    /* 
+     * The current state of the TSM. If the state is not TSM_READY, the remaining
+     * fields are invalid and will be initialized to 0.
+     */
+    enum tsm_state tsm_state;
+    /* Version number of the running TSM. */
+    unsigned long tsm_version;
+    /*
+     * The number of 4KiB pages which must be donated to the TSM for storing TVM
+     * state in sbi_tvm_create().
+     */
+    unsigned long vm_state_pages;
+    /* The maximum number of VCPUs a TVM can support. */
+    unsigned long tvm_max_vcpus;
+    /*
+     * The number of bytes per VCPU which must be donated to
+     * the TSM when creating a new TVM.
+     */
+    unsigned long tvm_bytes_per_vcpu;
+};
+------
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_tsm_get_info_errors]
+.TEE TSM Get Info
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | An input address was invalid.
+| SBI_ERR_INVALID_PARAM   | The input parameters were invalid.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE TSM Convert Pages (FID #1)
+[source, C]
+-----
+struct sbiret sbi_tee_tsm_convert_pages(unsigned long base_page_address,
+                                        unsigned long tsm_page_type,
+                                        unsigned long num_pages);
+
+-----
+Begins the process of converting `num_pages` of non-confidential memory starting
+at `base_page_address` to confidential-memory. On success, pages can be assigned
+to TVMs only following subsequent calls to `sbi_tee_tsm_initiate_fence()` and 
+`sbi_tee_tsm_local_fence()` that complete the conversion process.
+
+The `base_page_address` must be page-aligned. The `tsm_page_type` parameter must be
+a legal value for the `tsm_page_type` enum.
+
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_tsm_convert_pages_errors]
+.TEE TSM Convert Pages
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The operation completed successfully.
+| SBI_ERR_INVALID_PARAM | The input parameters were invalid.
+| SBI_ERR_FAILED        | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE TSM Initiate Fence (FID #2)
+[source, C]
+-----
+struct sbiret sbi_tee_tsm_initiate_fence(void);
+-----
+Initiates a TLB invalidation sequence for all pages marked for conversion via
+calls to `sbi_tee_tsm_convert_pages()` between the previous `sbi_tee_tsm_initiate_fence()`
+and now. The TLB invalidation sequence is completed when `sbi_tee_tsm_local_fence()` has
```

Comment:
Not sure what do you mean by "between the previous `sbi_tee_tsm_initiate_fence()`
and now." ?